### DISCUSSION TITLE
How to select a Postgres Schema?

### DISCUSSION THREAD

**Discussion by contibru** [quartznet/quartznet#2023] [Q&A]
**Describe the bug**

I have created all the quartz tables in a schema different from "public" and now when the application starts, it not recognizes the tables.
Is there a way to set the schema that quartz should use?

Error:
![image](https://user-images.githubusercontent.com/30509284/236949626-870297d6-2bba-44fc-94c4-b0b88fc5436b.png)

Quartz Configuration:

```
public static IServiceCollection AddQuartzServices(this IServiceCollection services, string connectionString)
        {
            services.AddQuartz(q =>
            {   
                q.UseMicrosoftDependencyInjectionJobFactory();
                q.UsePersistentStore(s =>
                {
                    s.PerformSchemaValidation = false; // default
                    s.UseProperties = true; // preferred, but not default
                    s.RetryInterval = TimeSpan.FromSeconds(15);
                    s.UsePostgres(postgres =>
                    {
                        postgres.ConnectionString = connectionString;
                    });
                    s.UseJsonSerializer();
                });
            });
            services.AddQuartzHostedService(opt =>
            {
                opt.WaitForJobsToComplete = true;
            });

            return services;
        }
```

**Version used**

3.6.2

**To Reproduce**

Ideally code samples or event unit test.

**Expected behavior**

**Additional context**

**Comment by lahma** [comment]
I believe that you should use table prefix configuration for this.

**Reply by contibru** [reply]
I have tried e still not working, I don`t believe this syntax is right for Postgres:

![image](https://github.com/quartznet/quartznet/assets/30509284/0c0af063-d1db-47bc-92be-3c02407faea2)

**Comment by jafin** [comment]
https://www.quartz-scheduler.net/documentation/quartz-3.x/configuration/reference.html#quartz-jobstore-tableprefix

**Reply by contibru** [reply]
I have tried e still not working, I don`t believe this syntax is right for Postgres:

![image](https://github.com/quartznet/quartznet/assets/30509284/0c0af063-d1db-47bc-92be-3c02407faea2)

**Comment by contibru** [comment]
@jafin and @lahma 
It worked without the "[]", I changed to :

`postgres.TablePrefix = "hh_tro_schd.qrtz_";`

Thanks for the help!
